### PR TITLE
Feature/autofill projectdetails

### PR DESCRIPTION
**Pull request by swarajsaaj**
Fix https://github.com/jhipster/jhipster-lite/issues/1858

### CODE REVIEW

File: src/main/webapp/app/springboot/domain/ProjectService.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import { Project } from '@/springboot/domain/Project';
 import { DocumentFile } from '@/common/domain/DocumentFile';
+import { RestProject } from '../secondary/RestProject';
```
Comment (pascalgrimaud) [inline src/main/webapp/app/springboot/domain/ProjectService.ts:3]: as we are in domain, we should not import a secondary

File: src/main/webapp/app/springboot/domain/ProjectService.ts [hunk 2]
```diff
@@ -12,4 +13,5 @@ export interface ProjectService {
   download(project: Project): Promise<DocumentFile>;
   addCodespacesSetup(project: Project): Promise<void>;
   addGitpodSetup(project: Project): Promise<void>;
+  getProjectDetails(folder: string): Promise<RestProject>;
```
Comment (pascalgrimaud) [inline src/main/webapp/app/springboot/domain/ProjectService.ts:16]: maybe return a Promise<Project> instead ?

**Review by pascalgrimaud — CHANGES_REQUESTED** [review]

File: src/main/webapp/app/springboot/domain/ProjectService.ts (hunk 2 above)
Comment (swarajsaaj) [reply src/main/webapp/app/springboot/domain/ProjectService.ts:16]: Do you mean `Promise<void>` instead of `Promise<RestProject>?`

Comment (pascalgrimaud) [reply src/main/webapp/app/springboot/domain/ProjectService.ts:16]: Yes, or Promise<Project>

Comment (swarajsaaj) [reply src/main/webapp/app/springboot/domain/ProjectService.ts]: Sure, as return type generic is not required for the feature, I have made it `Promise<void>` for generic required in promise

File: src/main/webapp/app/springboot/domain/ProjectService.ts (hunk 1 above)
Comment (swarajsaaj) [reply src/main/webapp/app/springboot/domain/ProjectService.ts]: Fixed in [727b697](https://github.com/jhipster/jhipster-lite/pull/1876/commits/727b6976e67a31641da8a03e8c794aa1831c755b)

**Review by pascalgrimaud — APPROVED** [review]

**Comment by swarajsaaj** [comment]
bounty claimed at https://opencollective.com/generator-jhipster/expenses/81769

**Comment by pascalgrimaud** [comment]
@swarajsaaj : approved :)

**Pull request merged**